docs/packaging.rst

```diff
@@ -225,32 +225,77 @@ src/snek5000_canonical/Snakefile
 
 .. code-block:: python
 
-   import os
-
    import snek5000
    from snek5000_canonical.output import OutputCanonical as Output
 
 
    # Case name
    CASE = "canonical"
    CONFIG = Output.get_configfile()
-   ROOT = Output.get_root()
-   NEK_SOURCE_ROOT = snek5000.source_root()
+
+   snek5000.ensure_env()
+
 
    configfile: CONFIG
 
-   if os.getenv("SNEK_DEBUG"):
-       config["CFLAGS"] += " -O0 -g"
-       config["FFLAGS"] += " -O0 -g -ffpe-trap=invalid,zero,overflow,underflow -Wall"
 
-   if not os.getenv("NEK_SOURCE_ROOT"):
-       os.environ["NEK_SOURCE_ROOT"] = NEK_SOURCE_ROOT
+   # Necessary to pass configuration to other Snakemake modules
+   config.update(
+       {
+           "CASE": CASE,
+           "file": CONFIG,
+           "includes": " ".join(Output().fortran_inc_flags),
+           "objects": " ".join(Output().makefile_usr_obj),
+       }
+   )
+
+   snek5000.append_gcc_debug_flags(config)
+
+
+   # default rule
+   rule all:
+       input:
+           "nek5000",
+
+
+   # shorthand for mesh
+   rule mesh:
+       input:
+           f"{CASE}.re2",
+           f"{CASE}.ma2",
+
+
+   # compiler and run rules
+   # ======================
+   module compiler:
+       snakefile:
+           snek5000.get_asset("compiler.smk")
+       config:
+           config
+
+
+   use rule * from compiler
```

Comment:
Just a question: is there a more explicit way to "import" rules from a snakemake module ?

Reply:
Indeed they are like star imports in Python. I tried this and yes it is possible:
```sh
use rule compile, mpiexec, run, srun from compiler
```
This however is not possible:
```py
use rule compile, run, srun from compiler
```
because the rules `run` and `srun` are derived from `mpiexec`. It seems like it is a bit different from Python's import system.